libtransmission/utils.c

```diff
@@ -2254,3 +2254,45 @@ void tr_net_init(void)
         initialized = true;
     }
 }
+
+/// mime-type
+
+#include "mime-types.c"
+
+static int compareSuffix(void const* va, void const* vb)
+{
+    char const* suffix = va;
+    struct mime_type_suffix const* entry = vb;
+    return tr_strcmp0(suffix, entry->suffix);
+}
+
+char const* tr_get_mime_type_for_filename(char const* filename)
+{
+    char const* ret = NULL;
+
+    char const* in = strrchr(filename, '.');
+    if ((in != NULL) && (strlen(++in) <= MIME_TYPE_SUFFIX_MAXLEN))
```

Comment:
Nitpick: could short-circuit here and then you don't even need `ret` (`return info != NULL ? info->mime_type : NULL`).

Reply:
good catch. tweaked in fa8544d8f9b7f7e749ac0e4d6708aecbe348f1ce